integration/update_test.go

```diff
@@ -188,3 +188,56 @@ func TestMultiFlag(t *testing.T) {
 		}
 	})
 }
+
+func TestUpdateReplaceDocuments(t *testing.T) {
+	setup.SkipForTigris(t)
+
+	t.Parallel()
+
+	stat := bson.D{{"n", int32(1)}, {"nModified", int32(1)}, {"ok", float64(1)}}
+
+	for name, tc := range map[string]struct {
```

Comment:
If I'm not mistaken, in all those test cases we add a new key called `"replacement-value"`, but we don't really replace the current keys. Let's add a test where the key in the update doc is not `replacement-value` but something that already was in the initial doc. Also, let's add a case where the update doc provides an empty object for update.

Reply:
That `expected` document compared against the document that was retrieved from DB.
In `Replace` we have `bson.D{{"_id", "bool-true"}, {"v", true}}` that replaced with `bson.D{{"_id", "bool-true"}, {"replacement-value", int32(1)}}`. In the `ReplaceWithDotNotation` the whole `document-composite` is replaced with `bson.D{{"_id", "document-composite"}, {"replacement-value", int32(1)}}`.

The thing about replacement objects is that it would replace the whole document except the `_id`.

So the first case you described would result in the same thing that we have now. The second one is good one. I have to debug the error I get on it.

Reply:
So, the first case is possible:
```
		"ReplaceDotNotationWithEmptyDoc": {
			update: bson.D{
				{"q", bson.D{{"v.array.0", bson.D{{"$eq", int32(42)}}}}},
				{"u", bson.D{{}}},
			},
			expectedFilter: bson.D{{"_id", "document-composite"}},
			expected:       bson.D{{"_id", "document-composite"}},
		},
```

But we can't add it right now, as the empty doc would result in parsing error.

Reply:
I created the issue for that case → https://github.com/FerretDB/FerretDB/issues/1000

Reply:
What a number! `:envy:`

Reply:
@w84thesun hmm, I'd suggest keeping this test case commented in the current branch with TODO that refers to the ticket you created.

Reply:
Fixed.